jep/7/README.adoc

```diff
@@ -67,22 +69,68 @@ endif::[]
+link:https://github.com/jenkinsci/ruby-runtime-plugin[Ruby Runtime plugin] and
+link:https://github.com/jenkinsci/jenkins.py[Jenkins Python SDK] will be suspended.
+Distribution of plugins depending on them will be suspended until their dependency is removed.
 
 
 == Specification
 
-`ruby-runtime` will be added to the `artifact-ignores` file of the update center generator so it is no longer available for download on update sites.
-// Perhaps: Add security warning?
+Ruby runtime and Python SDK will be added to the `artifact-ignores` file of the update center generator, so it is no longer available for download on update sites.
+A warning will be added to the update center so that all users of the plugins receive a warning in advance.
+The plugins will be also marked for adoption so that users can step up and take over maintenance without a 2 weeks waiting delay.
 
-Plugins with a mandatory dependency on `ruby-runtime` will be added to the same file, as they will not be installable without `ruby-runtime` being available.
+Plugins with a mandatory dependency on the runtimes will be added to the same file, as they will not be installable without the respective runtimes being available.
+The multiple Git repositories holding the `ruby-runtime` and Python SDK source code will be archived on GitHub.
 
-The multiple Git repositories holding the `ruby-runtime` source code will be archived on GitHub.
+Support for Ruby Runtime in the Jenkins core will be removed in future Jenkins versions.
+This includes but not limited to:
 
-Workarounds in Jenkins core implemented specifically to enable `ruby-runtime` will be reviewed and considered for reversal.
-No future core development will consider the impact on `ruby-runtime` and plugins based on it, and bug reports to core and related components about their impact on `ruby-runtime` and plugins based on it will be closed as *Won't Fix*.
+* link:https://github.com/stapler/stapler/tree/master/jruby[Stapler JRuby library] bundled in the Jenkins core
+* Permit lists for link:/jep/200[JEP-200: Switch Remoting/XStream denylist to a permitlist]
+* XStream converters for JRuby components
 
+No future core development will consider the impact on Ruby/Python runtimes and plugins based on them.
+Bug reports to core and related components about their impact on `ruby-runtime` and plugins based on it will be closed as *Won't Fix*.
+
+=== Rollout plan
+
+At least 1 month before removal::
+
+* Announcement blogpost on jenkins.io with the user-focused overview of the change and the delcivery plan
+* Plugins are marked for adoption
+* Plugins will be deprecated in the Update site immediately after the announcement.
+  The deprecation notice will reference the announcement blogpost as justification.
+* Plugin users will have 1 month to bring up their concerns, if any.
+  They will have an option to step up and contribute to the plugins so that they are migrated to the standard Jenkins plugin runtime.
```

Comment:
I do not see any particular reason for even the one-month delay—you can use the plugins as they are, on existing core versions, until they are rewritten to Java—but OK.

Reply:
Time for the plugin management flows like CasC to adjust

Reply:
If you are generating a Docker image or whatever from scratch using downloads. Yes I suppose so.

Reply:
The blogpost draft describes it in more details, with many use-cases mentioned

Reply:
> If you are generating a Docker image or whatever from scratch using downloads. Yes I suppose so.
lots of people do this with the helm chart, and then the k8s image fails to startup, and we get blamed.